docs/contributing-to-airbyte/resources/qa-checks.md

```diff
@@ -0,0 +1,81 @@
+### Connectors must have a changelog entry for each version
+*Applies to the following connector languages: java, low-code, python*
+
+Each new version of a connector must have a changelog entry defined in the user facing documentation in `./docs/integrations/<connector-type>s/<connector-name>.md`.
+
+## 📝 Metadata
+
+### Connectors must have valid metadata.yaml file
+*Applies to the following connector languages: java, low-code, python*
+
+Connectors must have a `metadata.yaml` file at the root of their directory. This file is used to build our connector registry. Its structure must follow our metadata schema. Field values are also validated. This is to ensure that all connectors have the required metadata fields and that the metadata is valid. More details in this [documentation](https://docs.airbyte.com/connector-development/connector-metadata-file).
+
+## 📦 Packaging
+
+### Connectors must use Poetry for dependency management
+*Applies to the following connector languages: python, low-code*
+
+Connectors must use [Poetry](https://python-poetry.org/) for dependency management. This is to ensure that all connectors use a dependency management tool which locks dependencies and ensures reproducible installs.
+### Connectors must be licensed under MIT or Elv2
+*Applies to the following connector languages: java, low-code, python*
+
+Connectors must be licensed under the MIT or Elv2 license. This is to ensure that all connectors are licensed under a permissive license. More details in our [License FAQ](https://docs.airbyte.com/developer-guides/licenses/license-faq).
+### Connector license in metadata.yaml and pyproject.toml file must match
+*Applies to the following connector languages: python, low-code*
+
+Connectors license in metadata.yaml and pyproject.toml file must match. This is to ensure that all connectors are consistently licensed
+### Connector version must follow Semantic Versioning
+*Applies to the following connector languages: java, low-code, python*
+
+Connector version must follow the Semantic Versioning scheme. This is to ensure that all connectors follow a consistent versioning scheme. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) for more details.
+### Connector version in metadata.yaml and pyproject.toml file must match
+*Applies to the following connector languages: python, low-code*
+
+Connector version in metadata.yaml and pyproject.toml file must match. This is to ensure that connector release is consistent.
+### Python connectors must have PyPi publishing enabled
+*Applies to the following connector languages: python, low-code*
+
+Python connectors must have [PyPi](https://pypi.org/) publishing enabled in their `metadata.yaml` file. This is declared by setting `remoteRegistries.pypi.enabled` to `true` in metadata.yaml. This is to ensure that all connectors can be published to PyPi and can be used in `airbyte-lib`.
```

Comment:
Question: If this is a must, why don't we just remove `remoteRegistries.pypi.enabled` and release to pypi when it is python/low-code?

Reply:
Good question.  I like the explicitness of the metadata flag and the possibility to disable publishing easily if we ever need it.

Reply:
It makes sense to me. Could we solve this by having a default value but still having the flag so that we can explicitly disable it if necessary? I think this is very low priority though